File: src/main/java/com/yauhenl/neuro/ai/Perceptron.java
package com.yauhenl.neuro.ai;

class Perceptron {
    private float[] weights;

    Perceptron(int n, float[] w) {
        weights = new float[n];
        System.arraycopy(w, 0, weights, 0, n);
    }

    float compute(float[] inputs) {
        float sum = 0;
        for (int i = 0; i < weights.length; ++i) {
            sum += inputs[i] * weights[i];
        }
        return activate(sum);
    }

    private float activate(float sum) {
        return sum > 0 ? 1 : -1;
    }
}

File: src/main/kotlin/com/yauhenl/kotlin/neuro/ai/Perceptron.kt
package com.yauhenl.kotlin.neuro.ai

internal class Perceptron(n: Int, w: FloatArray) {
    private val weights: FloatArray = FloatArray(n)

    init {
        System.arraycopy(w, 0, weights, 0, n)
    }

    fun compute(inputs: FloatArray): Float {
        var sum = 0f
        for (i in weights.indices) {
            sum += inputs[i] * weights[i]
        }
        return activate(sum)
    }

    private fun activate(sum: Float): Float {
        return (if (sum > 0) 1 else -1).toFloat()
    }
}
